appinventor/buildserver/tests/com/google/appinventor/buildserver/YailEvalTest.java

```diff
@@ -605,6 +605,47 @@ public void testListGroup5() throws Throwable {
     }
   }
 
+  public void testForEachDict() throws Throwable {
+    /* test for_each_dict block */
+    String schemeInputString = "(begin " +
+        "(def x 0)" +
+        "(foreach y " +
+        " (let " +
+        "   ( " +
+        "    ($key " +
+        "     (call-yail-primitive yail-list-get-item " +
+        "      (*list-for-runtime* (lexical-value y) 1) '(list number) \"select list item\" " +
+        "     ) " +
+        "    ) " +
+        "    ($value " +
+        "     (call-yail-primitive yail-list-get-item " +
+        "      (*list-for-runtime* (lexical-value y) 2) '(list number) \"select list item\" " +
+        "     ) " +
+        "    ) " +
+        "   ) " +
+        "   (set-var! x " +
+        "    (call-yail-primitive + " +
+        "     (*list-for-runtime* (get-var x) (lexical-value $key) (lexical-value $value) )" +
+        "     '(number number number ) \"+\"" +
+        "    ) " +
+        "   ) " +
+        "  ) " +
+        "  (call-yail-primitive make-yail-dictionary " +
+        "   (*list-for-runtime* " +
+        "    (call-yail-primitive make-ke-dictionary-pair " +
+        "     (*list-for-runtime* 1 2 ) '(key any)  \"make a pair\" " +
+        "    ) " +
+        "    (call-yail-primitive make-ke-dictionary-pair " +
+        "     (*list-for-runtime* 3 4 ) '(key any)  \"make a pair\" " +
+        "    ) " +
+        "   )" +
+        "  ) '(pair pair ) \"make a dictionary\"" +
```

Comment:
This line needs to be up one. Right now it is outside of the `(let)` which is one reason why the test fails (the error is foreach expects 3 args but got 5).